encoding.h

```diff
@@ -166,11 +167,19 @@
 #define PTE_A     0x040 // Accessed
 #define PTE_D     0x080 // Dirty
 #define PTE_SOFT  0x300 // Reserved for Software
-
+#define PTE_PPN 0x3ffffffffffc00
```

Comment:
Why have you assigned fixed addr for the L1 pt root?
You can use the `&` sign to get the addr in the C code.